sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerFailoverService.cs

```diff
@@ -0,0 +1,262 @@
+                                    break;
+                            }
+                        }
+                        catch (JsonException jex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] JSON parse issue for failoverLedger element: {jex.Message}");
+#endif
+                            _ = jex; // suppress unused warning in non-DEBUG builds
+                        }
+                        catch (InvalidOperationException ioex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] Invalid operation while parsing failoverLedger element: {ioex.Message}");
+#endif
+                            _ = ioex; // suppress unused warning in non-DEBUG builds
+                        }
+
+                        if (!string.IsNullOrEmpty(failoverLedgerId))
+                        {
+                            Uri endpoint = new UriBuilder(primaryEndpoint) { Host = $"{failoverLedgerId}.{LedgerDomainSuffix}" }.Uri;
+                            endpoints.Add(endpoint);
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                // ignore entire parse failure
+            }
+            return endpoints;
+        }
+
+        private HttpMessage CreateFailoverRequest(Uri failoverUrl)
+        {
+            HttpMessage message = _pipeline.CreateMessage(new RequestContext(), ResponseClassifier200);
+            Request request = message.Request;
+
+            request.Method = RequestMethod.Get;
+
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(failoverUrl);
+            request.Uri = uri;
+
+            request.Headers.Add("Accept", "application/json");
+
+            return message;
+        }
+
+        private static bool IsRetriableFailure(RequestFailedException ex)
+        {
+            // Include 404 and specific UnknownLedgerEntry error code.
+            return ex.Status == 404 ||
+                   string.Equals(ex.ErrorCode, "UnknownLedgerEntry", StringComparison.OrdinalIgnoreCase) ||
+                   ex.Status >= 500 ||
+                   ex.Status == 408 ||
+                   ex.Status == 429 ||
+                   ex.Status == 503 ||
+                   ex.Status == 504;
+        }
+
+        // Execute an operation only against discovered failover endpoints (skips primary). Used for specialized fallback flows.
+        public async Task<T> ExecuteOnFailoversAsync<T>(
+            Uri primaryEndpoint,
+            Func<Uri, Task<T>> operationAsync,
+            string operationName,
+            CancellationToken cancellationToken = default)
+        {
+            List<Uri> endpoints = await GetFailoverEndpointsAsync(primaryEndpoint, cancellationToken).ConfigureAwait(false);
+            Exception last = null;
+            foreach (var ep in endpoints)
```

Comment:
This is an interesting point. It is saying that the order of the failover ledgers is important and sequential. For instance if there are 2 failover ledgers and the primary fails it will redirect all traffic to the first failover ledger and then to third only if the primary and secondary fail. Third one would rarely be used in this case.

If the order does not matter much it might make sense to forward requests to a randomly ordered list of failover ledgers to improve the distribution of the requests. So that when we get x million of requests they do not suddenly get all forwarded to the failover ledger but distribute that load across all of the failover ones.